feat: add email template specification in doc (#898)
ory/kratos: docs/versioned_docs/version-v0.5/concepts/email-sms.md

    connection_uri: smtps://test:test@my-smtp-server:1025/
```

> ### Recipient and Templates

Reviewer: ```suggestion
### Sender Address and Template Customization
```

> You can custom recipient address and email templates.

Reviewer: ```suggestion
You can customize the sender address and email templates.
```

  #
  # Examples:
  # - /conf/courier-templates
  #
  # Set this value using environment variables on
  # - Linux/macOS:
  #    $ export COURIER_TEMPLATE_OVERRIDE_PATH=<value>
  # - Windows Command Line (CMD):
  #    > set COURIER_TEMPLATE_OVERRIDE_PATH=<value>
  #
  template_override_path: /conf/courier-templates
```

`email.subject.gotmpl` and `email.body.gotmpl` are common template file names expected in remainder directories corresponding to respective methods for filling E-mail subject and body.

> Templates use the engine golang text template for text/html email rendering: https://golang.org/pkg/text/template

- recovery: recovery email templates root directory
  - valid: sub directory containing templates with variables `To` and `VerificationURL` for validating a recovery
  - invalid: sub directory containing templates with variables `To` for invalidating a recovery
- verification: verification email templates root directory
  - valid: sub directory containing templates with variables `To` and `RecoveryURL` for validating a verification
  - invalid: sub directory containing templates with variables `To` for invalidating a verification

> For example: <https://github.com/ory/kratos/blob/master/courier/template/templates/verification/valid/email.body.gotmpl>

Reviewer: ```suggestion
For example: [`/courier/template/templates/verification/valid/email.body.gotmpl`](https://github.com/ory/kratos/blob/master/courier/template/templates/verification/valid/email.body.gotmpl)
```

> ```gotmpl title="verifiaction/valid/email.body.gotmpl"

Reviewer: ````suggestion
```gotmpl title="courier/template/templates/verification/valid/email.body.gotmpl"
````

Hi, please verify your account by clicking the following link:

<a href="{{ .VerificationURL }}">{{ .VerificationURL }}</a>
```

> > A future version of ORY Kratos will feature proprietary E-Mail messages using
> the Go template engine extended by
> [sprig's template functions](http://masterminds.github.io/sprig/). It should
> also be possible to internationalize these templates.

Reviewer: ```suggestion
```
